### PR TITLE
Indexed rendering

### PR DESCRIPTION
**Pull request by LucvandenBrand**
This fixes the low FPS count we had before.

It introduces a raw data store as a generic centralized location for us to put data.
It introduces a model factory, which parses raw model data and turns it into platform independent structs representing indexed vertex data.

Future improvements this leaves open: material properties are not parsed and the currently broken vertex shading (Suzanne is all white) could probably  be solved by providing the right data too.

### CODE REVIEW

File: app/src/raw_data_store.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+/**
+ * Data structure for the raw data store
+ */
+pub struct RawDataStore {
+    raw_data_map: BTreeMap<&'static str, &'static [u8]>,
+}
+
+/**
+ * Implementation of the raw data store: allows for fast searching.
+ */
```
Comment (Qqwy) [inline app/src/raw_data_store.rs:23]: Good idea, and good simple implementation 👍

**Review by Qqwy — APPROVED** [review]
This looks very good 👍 . I do not have remarks w.r.t. the code, and it also runs very fast on my machine :-)

File: app/src/renderer/indexed_model.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ */
+impl IndexedModel {
+    /**
+     * Turn a model into its indexed equivalent
+     */
+    pub fn new(object: &Obj) -> IndexedModel {
+        let mut vertex_map: Vec<Vertex> = Vec::new();
+        let mut positions: Vec<f32> = Vec::new();
+        let mut indices: Vec<u16> = Vec::new();
+        for vertex in object.vertices() {
```
Comment (Qqwy) [inline app/src/renderer/indexed_model.rs:24]: There might be a way to make the implementation of this loop cleaner. (like by using only iterator functions and no mutable variables) but I do not think the extra time spent writing that is worth it.

File: app/src/raw_data_store.rs (hunk 1 above)
Comment (LucvandenBrand) [reply app/src/raw_data_store.rs:23]: Thanks :)

File: app/src/renderer/indexed_model.rs (hunk 2 above)
Comment (LucvandenBrand) [reply app/src/renderer/indexed_model.rs:24]: I'll see if I can improve this when I start work on textured rendering next :)

**Pull request merged**